### PR TITLE
Fix native build

### PR DESCRIPTION
**Pull request by arichardson**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -140,8 +144,9 @@ if(NOT DEFINED SNMALLOC_ONLY_HEADER_LIBRARY)
     target_compile_definitions(${name} PRIVATE "SNMALLOC_EXPORT=__attribute__((visibility(\"default\")))")
     set_target_properties(${name} PROPERTIES CXX_VISIBILITY_PRESET hidden)
 
-    if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^CHERI")
+    if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^CHERI" AND USE_REVOCATION)
       target_link_libraries(${name} cheri_caprevoke)
+      target_compile_definitions(${name} PRIVATE -DSNMALLOC_REVOKE_QUARANTINE=1)
```
Comment (arichardson) [inline CMakeLists.txt]: Should this also set SNMALLOC_QUARANTINE_DEALLOC?

Comment (nwf) [reply CMakeLists.txt]: Probably yes.  I've gone back and forth about what the right way to set these knobs for the purposes of experimentation is and have defaulted to just *checking* or passing through and leaving the actual "do the right thing" to cheribuild.  But for people without cheribuild, this is probably the right place to do the right thing.

**Comment by nwf** [comment]
I've pulled these in as part of my most recent rebase.  If you're good with that, this can be closed.

**Comment by arichardson** [comment]
Thanks!

**Pull request closed without merging**